### PR TITLE
[SPARK-21805][SPARKR] Disable R vignettes code on Windows

### PR DESCRIPTION
**Pull request by felixcheung**
## What changes were proposed in this pull request?

Code in vignettes requires winutils on windows to run, when publishing to CRAN or building from source, winutils might not be available, so it's better to disable code run (so resulting vigenttes will not have output from code, but text is still there and code is still there)

fix * checking re-building of vignette outputs ... WARNING
and
> %LOCALAPPDATA% not found. Please define the environment variable or restart and enter an installation path in localDir.

## How was this patch tested?

jenkins, appveyor, r-hub

before: https://artifacts.r-hub.io/SparkR_2.2.0.tar.gz-49cecef3bb09db1db130db31604e0293/SparkR.Rcheck/00check.log
after: https://artifacts.r-hub.io/SparkR_2.2.0.tar.gz-86a066c7576f46794930ad114e5cff7c/SparkR.Rcheck/00check.log

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by felixcheung** [comment]
what's with jenkins? looks like JIRA is not updated either...

**Comment by HyukjinKwon** [comment]
Yes ... for JIRA linking problem, there are some information here - http://apache-spark-developers-list.1001551.n3.nabble.com/Some-PRs-not-automatically-linked-to-JIRAs-td22067.html just FYI. I think -9 has increased recently..

**Comment by shivaram** [comment]
Thanks @felixcheung ! Are the warnings about the missing PDF unavoidable ? I see something like
```
 checking package vignettes in 'inst/doc' ... WARNING
Package vignette without corresponding PDF/HTML:
   'sparkr-vignettes.Rmd'
```

**Comment by felixcheung** [comment]
I think that is caused by the way I build the source package - I use the option --no-build-vignettes so the source package I test with doesn't have the PDF in it

Similarly for the failed test - I change the version to 2.2.0 to get it to run without setting the download URL and there is a change on the wrapper class parameter list.

**Comment by shivaram** [comment]
Ah I see. Yeah the failed tests makes sense. We can also try to submit a custom tar.gz to r-hub to test it with the PDF and a different version number ?

**Comment by felixcheung** [comment]
It likely best to back port this to branch-2.2 so the test can pass with matching API in a released jar.

Then we can see everything passing (except having to hand edit the version number)

File: R/pkg/vignettes/sparkr-vignettes.Rmd [hunk 1]
```diff
@@ -27,6 +27,17 @@ vignette: >
  limitations under the License.
 -->
 
+```{r setup, include=FALSE}
+library(knitr)
```
Comment (shivaram) [inline R/pkg/vignettes/sparkr-vignettes.Rmd]: Just to clarify, the opt_hooks is provided by knitr ?

**Comment by shivaram** [comment]
Sure - change LGTM. Lets see if @HyukjinKwon has any more comments ? If not we can merge to master, branch-2.2 and then do some more tests.

File: R/pkg/vignettes/sparkr-vignettes.Rmd (hunk 1 above)
Comment (felixcheung) [inline R/pkg/vignettes/sparkr-vignettes.Rmd]: yes!

**Comment by HyukjinKwon** [comment]
Sure, I have no more comments and it looks good to me too.

**Comment by felixcheung** [comment]
thanks, merged to master/2.2.
will check for nightly build from tonight.

**Comment by felixcheung** [comment]
I tested with the nightly, it passed
https://builder.r-hub.io/status/SparkR_2.2.0.tar.gz-fba296d2142edeb8bcdb082e311d8462

**Comment by shivaram** [comment]
Thats great ! I will also run this by winbuilder later today.

**Pull request closed without merging**